clang/include/clang/Basic/LangOptions.h

```diff
@@ -647,6 +647,14 @@ class LangOptions : public LangOptionsBase {
     return ConvergentFunctions;
   }
 
+  /// Return true if atomicrmw operations targeting allocations in private
```

Comment:
Do we want to have a check in target machine to tell if atomic operation on specific address space is legal? I'm thinking of adding atomic support in `AAAddressSpace`, and could drop the address space if an atomic operation in an inferred address space is not legal.

Reply:
This has nothing to do with the target, this is language semantics. What the target natively supports is an incomprehensibly complicated question, and is a legalization issue. We could annotate some operations with this in infer address spaces, but there's only a few cases where it's probably useful there